pkg/controller/keyvault/kv_util.go

```diff
@@ -16,40 +15,60 @@ import (
 
 var nginxNamePrefix = "keyvault-nginx-"
 
-func buildSPC(obj client.Object, spc *secv1.SecretProviderClass, config *config.Config) (bool, error) {
-	var certURI, specSecretName string
+type EventType int
 
+const (
+	Normal EventType = iota
+	Warning
+)
+
+func (et EventType) String() string {
+	switch et {
+	case Warning:
+		return "Warning"
+	default:
+		return "Normal"
+	}
+}
+
+type SPCConfig struct {
+	ClientId        string
+	TenantId        string
+	KeyvaultCertUri string
+	Name            string
+	Cloud           string
+}
+
+func shouldDeploySpc(obj client.Object) bool {
```

Comment:
this abstraction doesn't work for gateway?

Reply:
No since SPCs in Gateways map to individual listeners, which don't satisfy the obj interface.

Rather than have one method that does two things, determine if an SPC should be deployed AND build the SPC itself, I figured it would make sense to break that up into two separate methods, which is what I do for a per-listener basis on each Gateway resource we reconcile. 

This way, we can keep the process of building SPCs the same, whether the SPC is based on an individual listener or on an ingress/nic - the SPCConfig object provides an abstraction for all three to work